lib/postcode_validation/gateway/pca_address_list.rb

```diff
@@ -0,0 +1,43 @@
+module PostcodeValidation
+  module Gateway
+    class PCAAddressList
+      include HTTParty
+
+      class PCARequestError < PostcodeValidation::Error::RequestError; end
+
+      KEY = ENV['POSTCODE_ANYWHERE_KEY']
+      base_uri 'https://services.postcodeanywhere.co.uk'
```

Comment:
You could use a strategy pattern, so when called from the use cases

```ruby
gateway.query(search_term: postcode, country: country, querying_strategy: ShortList.new)
```

```ruby
gateway.query(search_term: postcode, country: country, querying_strategy: FullListing.new)
```

Inside the gateway the strategy would contain the logic that is different:

```ruby
url = strategy.url
filter = strategy.filter #None vs Postcode
```